include/nbl/builtin/hlsl/scan/default_scheduler.hlsl

```diff
@@ -35,187 +17,181 @@ namespace scan
-	 * |> finishedFlagOffset - an index in the scratch buffer where each virtual workgroup indicates that ALL its invocations have finished their work. This helps 
-	 *							synchronizing between workgroups with while-loop spinning.
-	 */
-	void computeParameters(in uint elementCount, out Parameters_t _scanParams, out DefaultSchedulerParameters_t _schedulerParams)
-	{
+    /**
+     * The CScanner.h parameter computation calculates the number of virtual workgroups that will have to be launched for the Scan operation
+     * (always based on the elementCount) as well as different offsets for the results of each step of the Scan operation, flag positions 
+     * that are used for synchronization etc.
+     * Remember that CScanner does a Blelloch Scan which works in levels. In each level of the Blelloch scan the array of elements is 
+     * broken down into sets of size=WorkgroupSize and each set is scanned using Hillis & Steele (aka Stone-Kogge adder). The result of 
+     * the scan is provided as an array element for the next level of the Blelloch Scan. This means that if we have 10000 elements and 
+     * WorkgroupSize=250, we will break the array into 40 sets and take their reduction results. The next level of the Blelloch Scan will 
+     * have an array of size 40. Only a single workgroup will be needed to work on that. After that array is scanned, we use the results 
+     * in the downsweep phase of Blelloch Scan.
+     * Keep in mind that each virtual workgroup executes a single step of the whole algorithm, which is why we have the cumulativeWorkgroupCount.
+     * The first virtual workgroups will work on the upsweep phase, the next on the downsweep phase.
+     * The intermediate results are stored in a scratch buffer. That buffer's size is is the sum of the element-array size for all the 
+     * Blelloch levels. Using the previous example, the scratch size should be 10000 + 40.
+     * 
+     * Parameter meaning:
+     * |> lastElement - the index of the last element of each Blelloch level in the scratch buffer
+     * |> topLevel - the top level the Blelloch Scan will have (this depends on the elementCount and the WorkgroupSize)
+     * |> temporaryStorageOffset - an offset array for each level of the Blelloch Scan. It is used when storing the REDUCTION result of each workgroup scan
+     * |> cumulativeWorkgroupCount - the sum-scan of all the workgroups that will need to be launched for each level of the Blelloch Scan (both upsweep and downsweep)
+     * |> finishedFlagOffset - an index in the scratch buffer where each virtual workgroup indicates that ALL its invocations have finished their work. This helps 
+     *                            synchronizing between workgroups with while-loop spinning.
+     */
+    void computeParameters(NBL_CONST_REF_ARG(uint32_t) elementCount, out Parameters_t _scanParams, out DefaultSchedulerParameters_t _schedulerParams)
+    {
```

Comment:
The parameters are "married" to to the scheduling logic.

So make these methods of `DefaultSchedulerParameters_t` , this one is really just a constructor so turn it into a `static DefaultSchedulerParameters_t create(...)` method

Hint: whenever a function takes `inout` or `out` struct `T` as first parameter, it likely means that I wrote it in GLSL to be a pseudo-method (would have a C fn ptr signature matching a method). 

Bonus Hint: Whenever the first param is a `in T` it would have been a const method in C++.